Sources/SwiftParser/Lookahead.swift

```diff
@@ -27,22 +27,30 @@ extension Parser {
     /// i.e. how far it looked ahead.
     var tokensConsumed: Int = 0
 
+    var parseTransition: IncrementalParseTransition?
```

Comment:
As discussed offline, we could store `furthestOffset` in a standalone `struct` and pass that over here as an `UnsafeMutablePointer`.

Reply:
Since we have to record the offset in both `Parser.peek()` and `Lookahead.consumeAnyToken()`, I make this pointer a member of `LexemeSequence`.